airflow/models/dagrun.py

```diff
@@ -451,13 +452,18 @@ def get_running_dag_runs_to_examine(cls, session: Session) -> Query:
             .order_by(
                 nulls_first(BackfillDagRun.sort_ordinal, session=session),
                 nulls_first(cls.last_scheduling_decision, session=session),
-                cls.logical_date,
+                cls.run_after,
             )
             .limit(cls.DEFAULT_DAGRUNS_TO_EXAMINE)
         )
 
         if not settings.ALLOW_FUTURE_LOGICAL_DATES:
-            query = query.where(DagRun.logical_date <= func.now())
+            query = query.where(
+                or_(
+                    DagRun.logical_date.is_(null()),
```

Comment:
I don’t even know there’s `null()`… None works elsewhere. (We don’t need to change this.)

Reply:
This will be removed after your PR is merged

Reply:
updated